tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_BjorhusImpl.cpp

```diff
@@ -0,0 +1,848 @@
+          outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+          projection_ab, projection_Ab, projection_AB, inverse_spatial_metric,
+          extrinsic_curvature, spacetime_metric, inverse_spacetime_metric,
+          three_index_constraint, char_projected_rhs_dt_v_psi,
+          char_projected_rhs_dt_v_minus, constraint_char_zero_plus,
+          constraint_char_zero_minus, phi, d_phi, d_pi, char_speed_array);
+  return dt_v_minus;
+}
+
+template <size_t VolumeDim>
+void test_add_constraint_preserving_terms_to_dt_v_minus(
+    const size_t grid_size_each_dimension) noexcept {
+  pypp::check_with_random_values<1>(
+      &wrapper_func_cp_v_minus<VolumeDim>,
+      "Evolution.Systems.GeneralizedHarmonic.BoundaryConditions.Bjorhus",
+      "add_constraint_preserving_terms_to_dt_v_minus", {{{-1., 1.}}},
+      DataVector(grid_size_each_dimension));
+}
+
+template <size_t VolumeDim>
+void test_add_constraint_preserving_physical_terms_to_dt_v_minus(
+    const size_t grid_size_each_dimension) noexcept {
+  pypp::check_with_random_values<1>(
+      &wrapper_func_cpp_v_minus<VolumeDim>,
+      "Evolution.Systems.GeneralizedHarmonic.BoundaryConditions.Bjorhus",
+      "add_constraint_preserving_physical_terms_to_dt_v_minus", {{{-1., 1.}}},
+      DataVector(grid_size_each_dimension));
+}
+}  // namespace
+
+namespace {
+using frame = Frame::Inertial;
+constexpr size_t VolumeDim = 3;
+
+void test_constraint_preserving_physical_bjorhus_v_minus_vs_spec_3d(
+    const size_t grid_size_each_dimension,
+    const std::array<double, 3>& lower_bound,
+    const std::array<double, 3>& /* upper_bound */) noexcept {
+  // Setup grid
+  Mesh<VolumeDim> mesh{grid_size_each_dimension, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  // Setup coordinates
+  const Direction<VolumeDim> direction(1, Side::Upper);  // +y direction
+  const size_t slice_grid_points =
+      mesh.extents().slice_away(direction.dimension()).product();
+  const auto inertial_coords = [&slice_grid_points, &lower_bound]() {
+    tnsr::I<DataVector, VolumeDim, frame> tmp(slice_grid_points, 0.);
+    // +y direction
+    get<1>(tmp) = 0.5;
+    for (size_t i = 0; i < VolumeDim; ++i) {
+      for (size_t j = 0; j < VolumeDim; ++j) {
+        get<0>(tmp)[i * VolumeDim + j] =
+            lower_bound[0] + 0.5 * static_cast<double>(i);
+        get<2>(tmp)[i * VolumeDim + j] =
+            lower_bound[2] + 0.5 * static_cast<double>(j);
+      }
+    }
+    return tmp;
+  }();
+
+  // Populate various tensors needed to compute BcDtVMinus as done in SpEC
+  tnsr::I<DataVector, VolumeDim, frame> local_unit_interface_normal_vector(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  tnsr::I<DataVector, VolumeDim, frame> local_inertial_coords(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  const auto& lic0 = get<0>(local_inertial_coords);
```

Comment:
can you name these something easier to understand than `lic`? Eg `local_coords_x`. If there aren't any grid/logical coords I'd be happy with you omitting the `inertial` parts if you want to save typing

Reply:
renamed to `local_inertial_coords_x,y,z` :)